### PR TITLE
Kick user to the screensaver and reset app upon idle

### PR DESCRIPTION
**Pull request by fungjj92**
## Overview

When the app goes idle, we want to reset the app state and display the screensaver. Furthermore, a video playing should not count as idle time. This PR adds this timer via[ react-idle-timer ](https://github.com/SupremeTechnopriest/react-idle-timer)and sets it up to be controlled by videos across the app.

The timer is set to time out after 5 seconds in this PR. In production that number will be much higher (a minute?).

~The two actions will fit in nicely with how videos are manually controlled in #50 . Playing a video should call `pauseTimer`. A video stopping manually should call `resetTimer`. TBD how to detect a video stopping for other reasons like playing through to the end. When switching tabs suddenly, we can hook into unmounting lifecycle hooks.~ Implemented because #50 was merged. Open to rethinking this workflow, because Redux doesn't feel like it was built to solve this problem -- but using it does the job.

Connects #16 

### Notes

Before merging:
- I will drop c48aaff -- I did that work to test that hooking into the timer works from any other component.
- I will up the time out so it's not so much in the way during development. It can easily be turned off or bumped up in time.

## Testing Instructions
`./scripts/update` if testing locally
Click into the app and test the app goes to screensaver after 5 seconds.
The timer should reset back to 5s with any touch or movement.
Interact with the buttons on the quiz tab and see the app behaves as expected
Interact with the videos and switch tabs and see that the timeout behaves as expected.

### CODE REVIEW

**Comment by fungjj92** [comment]
Hold off review until #50 is merged.

**Comment by maurizi** [comment]
> TBD how to detect a video stopping for other reasons like playing through to the end.

I think we can use the `ended` event for this. Adding an `onEnded` callback to the `video` element which calls `resetTimer` should suffice.
https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/ended_event

File: src/app/src/App.js
Comment (maurizi) [inline src/app/src/App.js]: Perhaps this is a lack of familiarity with React, but does this need to be `shouldComponentUpdate`?

We always return `true` here, so can't we use `componentDidUpdate` instead?

File: src/app/src/App.js
Comment (maurizi) [inline src/app/src/App.js]: Consider pulling this out to a module-level constant or moving it to `constants.js`.

**Review by maurizi — APPROVED** [review]
:+1: I left a few comments, but feel free to disregard them if you disagree - I don't feel that strongly about them.

File: src/app/src/App.js
Comment (fungjj92) [reply src/app/src/App.js]: Good suggestion!

**Comment by fungjj92** [comment]
Thanks for review!

**Pull request merged**
